File: BACKEND/ColegioMaster.Negocio/Juegos/JuegoRecreativoService.cs
using ColegioMaster.DtoModels.JuegoRecreativo;
using ColegioMaster.Repositorio.Implement;

namespace ColegioMaster.Negocio.Juegos
{
    public class JuegoRecreativoService : IJuegoRecreativoService
    {
        private readonly IJuegoRecreativoRepositorio _repository;

        public JuegoRecreativoService(IJuegoRecreativoRepositorio repository)
        {
            _repository = repository;
        }

        public async Task<JuegoRecreativoDto> Create(JuegoRecreativoCrearDto request)
        {
            JuegoRecreativoDto result = await _repository.Create(request);
            return result;
        }

        public async Task<bool> Delete(int id)
        {
            bool result = await _repository.Delete(id);
            return result;
        }

        public async void Dispose()
        {
            GC.SuppressFinalize(this);
        }

        public async Task<List<JuegoRecreativoDto>> GetAll()
        {
            List<JuegoRecreativoDto> result = await _repository.GetAll();
            return result;
        }

        public async Task<JuegoRecreativoDto> GetById(int id)
        {
            JuegoRecreativoDto result = await _repository.GetById(id);
            return result;
        }

        public async Task<JuegoRecreativoDto> Update(int id, JuegoRecreativoActualizarDto request)
        {
            JuegoRecreativoDto result = await _repository.Update(id, request);
            return result;
        }
    }
}

File: FrontEnd/src/app/services/JuegoRecreativo/JuegoRecreativo.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { apiEndpoints } from '../../shared';

export type JuegoRecreativoDto = {
  id: number;
  nombre: string;
  tipo?: string;
  estado?: string;
  fechaCreacion?: string;
  fechaModificacion?: string;
  usuarioCreacion?: string;
  usuarioModificacion?: string;
};

export type JuegoRecreativoCrearDto = {
  nombre: string;
  tipo?: string;
  estado?: string;
  usuarioCreacion: string;
};

export type JuegoRecreativoActualizarDto = {
  nombre: string;
  tipo?: string;
  estado?: string;
  usuarioModificacion?: string;
};

@Injectable({
  providedIn: 'root'
})
export class JuegoRecreativoService {
  private readonly _http = inject(HttpClient);
  private readonly apiUrl = `${apiEndpoints.juegoRecreativo}`;

  constructor() { }

  getAll(): Observable<JuegoRecreativoDto[]> {
    return this._http.get<JuegoRecreativoDto[]>(this.apiUrl);
  }

  getById(id: number): Observable<JuegoRecreativoDto> {
    return this._http.get<JuegoRecreativoDto>(`${this.apiUrl}/${id}`);
  }

  create(payload: JuegoRecreativoCrearDto): Observable<JuegoRecreativoDto> {
    return this._http.post<JuegoRecreativoDto>(this.apiUrl, payload);
  }

  update(id: number, payload: JuegoRecreativoActualizarDto): Observable<JuegoRecreativoDto> {
    return this._http.put<JuegoRecreativoDto>(`${this.apiUrl}/${id}`, payload);
  }

  delete(id: number): Observable<boolean> {
    return this._http.delete<boolean>(`${this.apiUrl}/${id}`);
  }
}